src/ast/nodes/ExportDefaultDeclaration.ts

```diff
@@ -159,7 +159,7 @@ export default class ExportDefaultDeclaration extends NodeBase {
 		);
 		const hasTrailingSemicolon = code.original.charCodeAt(this.end - 1) === 59; /*";"*/
 		if (systemBinding) {
-			code.appendLeft(
```

Comment:
It's particularly the system binding case that suffers from the double write issue, where we want to add an appendage first, then later on rewrite the inner content. It might be more obvious think in terms of `overwriteInner` and `overwriteOuter` here, see https://github.com/Rich-Harris/magic-string/pull/144#issuecomment-393532572 for further comment on this.